### PR TITLE
feature: compatible with redisson versions lower than 3.16.0(<3.16.0)

### PR DESCRIPTION
**Pull request by DanLi39**

### CODE REVIEW

File: arex-instrumentation/redis/arex-redission-v3/src/main/java/io/arex/inst/redisson/v3/RedissonInstrumentation.java [hunk 1]
```diff
@@ -71,9 +79,9 @@ public static boolean onEnter() {
 
         @Advice.OnMethodExit(suppress = Throwable.class)
         public static <V> void onExit(@Advice.Argument(0) String name,
-            @Advice.FieldValue("commandExecutor") CommandAsyncExecutor commandExecutor,
+            @Advice.This RedissonClient redisson,
             @Advice.Return(readOnly = false) RBucket<V> redissonBucket) {
-            redissonBucket = new RedissonBucketWrapper<>(commandExecutor, name);
+            redissonBucket = new RedissonBucketWrapper<>(getCommandAsyncExecutor(redisson), name);
```
Comment (mr3) [inline arex-instrumentation/redis/arex-redission-v3/src/main/java/io/arex/inst/redisson/v3/RedissonInstrumentation.java:84]: Why not put `commandAsyncExecutorMap` in class RedissonHelper.java?

`RedissonInstrumentation` classloader is `AgentClassLoader`
`redisson` classloader is `UserClassLoader`

**Review by mr3 — APPROVED** [review]

**Review by mr3 — DISMISSED** [review]

File: arex-instrumentation/redis/arex-redission-v3/src/main/java/io/arex/inst/redisson/v3/common/RedissonHelper.java [hunk 2]
```diff
@@ -57,4 +60,31 @@ public static MasterSlaveServersConfig getMasterSlaveServersConfig(ConnectionMan
...
+     *
+     * @param redisson
+     * @return
+     */
+    public static CommandAsyncExecutor getCommandAsyncExecutor(RedissonClient redisson) {
+        if (redisson == null) {
+            return null;
+        }
+        if (commandAsyncExecutorMap.containsKey(redisson.hashCode())) {
+            return commandAsyncExecutorMap.get(redisson.hashCode());
```
Comment (lucas-myx) [inline arex-instrumentation/redis/arex-redission-v3/src/main/java/io/arex/inst/redisson/v3/common/RedissonHelper.java:74]:
```
CommandAsyncExecutor executor = commandAsyncExecutorMap.get(redisson.hashCode();
if (executor != null) {
    return executor;
}
```

File: arex-instrumentation/redis/arex-redission-v3/src/main/java/io/arex/inst/redisson/v3/common/RedissonHelper.java [hunk 3]
```diff
@@ -57,4 +60,31 @@ public static MasterSlaveServersConfig getMasterSlaveServersConfig(ConnectionMan
...
+        }
+        Method getCommandExecutor = ReflectUtil.getMethod(redisson.getClass(), "getCommandExecutor");
```
Comment (lucas-myx) [inline arex-instrumentation/redis/arex-redission-v3/src/main/java/io/arex/inst/redisson/v3/common/RedissonHelper.java:76]: static Method getCommandExecutorMethod = ReflectUtil.getMethod(redisson.getClass(), "getCommandExecutor");
if (getCommandExecutorMethod != null) {
// invoke
// put map
}

File: arex-instrumentation/redis/arex-redission-v3/src/main/java/io/arex/inst/redisson/v3/common/RedissonHelper.java [hunk 4]
```diff
@@ -2,18 +2,21 @@
...
 import org.redisson.connection.ConnectionManager;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 import io.arex.inst.runtime.log.LogManager;
 
 public class RedissonHelper {
 
     private static Method getConfigMethod = ReflectUtil.getMethod(ConnectionManager.class, "getConfig");
     private static Set<MasterSlaveServersConfig> masterSlaveServersConfigSet = new HashSet<>(1);
+    private static Map<Integer, CommandAsyncExecutor> commandAsyncExecutorMap = new HashMap<>();
```
Comment (lucas-myx) [inline arex-instrumentation/redis/arex-redission-v3/src/main/java/io/arex/inst/redisson/v3/common/RedissonHelper.java:19]: remove commandAsyncExecutorMap

**Review by lucas-myx — APPROVED** [review]

**Review by mr3 — APPROVED** [review]

**Pull request merged**